lightning/src/routing/scoring.rs

```diff
@@ -810,14 +829,30 @@ impl<L: Deref<Target = u64>, BRT: Deref<Target = HistoricalBucketRangeTracker>,
 			// sending less than 1/16th of a channel's capacity, or 1/8th if we used the top of the
 			// bucket.
 			let mut total_valid_points_tracked = 0;
+			let required_decays = self.now.duration_since(*self.last_updated).as_secs()
+				.checked_div(params.historical_no_updates_half_life.as_secs())
+				.map_or(u32::max_value(), |decays| cmp::min(decays, u32::max_value() as u64) as u32);
+
+			// Rather than actually decaying the individual buckets, which would lose precision, we
```

Comment:
Hmm, how does this lose precision? Seems to be the case that we might treat mostly-decayed data as not decayed at all?